spec/factories/uploads.rb

```diff
@@ -10,15 +10,11 @@
     is_consistent { true }
 
     trait :with_chunks do
-      chunks { [ build(:chunk, number: 1) ] }
+      chunks { [ create(:chunk, :skip_validation, number: 1) ] }
```

Comment:
I don't like calling `create` on traits that add child objects. In this case, if you want call `build( :upload, :with chunks)`, the chunks will be persisted and then added to the build (but not persisted) upload. I'm /guessing/ that what was causing problems here is that, even when calling `build` on a chunk, FactoryBot would try to create an upload for it. What would probably fix that issue is to pass `upload: nil` like:
```suggestion
      chunks { [ build(:chunk, upload: nil, number: 1) ] }
```
Please let me know if I'm completely off base. Hopefully we can find a solution that makes us both happy.

Reply:
I think this is a problem with the way we use :with_chunks.  Most of our test create(:upload, :with_chunks). If I pass upload: nil then all of these creates will fail because the resulting upload.chunks would be invalid (they need an upload). The few places were we build an upload, we should probably just build chunks explicitly.

Reply:
fixed